src/de/grobox/liberario/fragments/DirectionsFragment.java

```diff
@@ -443,29 +444,59 @@ private void processIntent() {
 		}
 	}
 
+	private void presetFromTo(WrapLocation wfrom, WrapLocation wvia, WrapLocation wto, Date date) {
+
+		Location from, via, to;
+		if(wfrom != null) {
+			from = wfrom.getLocation();
+			if(wfrom.getType() == WrapLocation.WrapType.GPS) {
+				activateGPS();
+				from = null;
+			}
+		} else {
+			from = null;
+		}
+		if(wvia != null) {
+			via = wvia.getLocation();
+		} else {
+			via = null;
+		}
+		if(wto != null) {
+			to = wto.getLocation();
+			if(wto.getType() == WrapLocation.WrapType.HOME){
+				to = null;
+				ui.to.setWrapLocation(wto);
+			}
+		} else {
+			to = null;
+		}
+
+		presetFromTo(from, via, to, date);
+	}
+
 	private void presetFromTo(Location from, Location via, Location to, Date date) {
```

Comment:
I could do that but that would make the method `presetFromTo()` more complex. I will refactor this and then we can see.